File: programmers/[고득점Kit][브루트포스]/카펫.py
# 카펫

def solution(brown, yellow):
    y = 3
    x = 0

    while True:
        x = (yellow / (y - 2)) + 2
        x = int(x)
        if yellow == (x - 2) * (y - 2):
            if brown == (x + y - 2) * 2:
                break
        y += 1

    return [x, y]

File: By-Swift/By-Swift/Programmers/수학/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    var y = 3
    var x = 0
    
    while true {
        x = Int((yellow + 2*y - 4) / (y-2))
        if yellow == (x*y - 2*(x+y-2)) {
            if brown == 2*(x+y-2) {
                break
            }
        }
        y += 1
    }
    return [x, y]
}